Database constraints use deprecated check kwarg

Django 5.1 deprecated the CheckConstraint 'check' parameter in favor of 'condition', and [Django 6.0 (released December 3, 2025)](https://docs.djangoproject.com/en/6.0/releases/6.0/) removed it entirely.

This was causing the Sphinx documentation build to fail because the docs tox environment has no upper bound on Django versions (django>=4.2.0), so it installed Django 6.0 which no longer accepts the 'check' parameter:

```
       File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/.tox/docs/lib/python3.12/site-packages/sphinx/config.py", line 611, in eval_config_file
        raise ConfigError(msg % traceback.format_exc()) from exc
    sphinx.errors.ConfigError: There is a programmable error in your configuration file:
    
    Traceback (most recent call last):
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/.tox/docs/lib/python3.12/site-packages/sphinx/config.py", line 596, in eval_config_file
        exec(code, namespace)  # NoQA: S102
        ^^^^^^^^^^^^^^^^^^^^^
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/docs/conf.py", line 26, in <module>
        django.setup()
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/.tox/docs/lib/python3.12/site-packages/django/__init__.py", line 24, in setup
        apps.populate(settings.INSTALLED_APPS)
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/.tox/docs/lib/python3.12/site-packages/django/apps/registry.py", line 116, in populate
        app_config.import_models()
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/.tox/docs/lib/python3.12/site-packages/django/apps/config.py", line 269, in import_models
        self.models_module = import_module(models_module_name)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.12/x64/lib/python3.12/importlib/__init__.py", line 90, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
      File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
      File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 999, in exec_module
      File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/digid_eherkenning/models/__init__.py", line 3, in <module>
        from .eherkenning import EherkenningConfiguration
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/digid_eherkenning/models/eherkenning.py", line 39, in <module>
        class EherkenningConfiguration(BaseConfiguration):
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/digid_eherkenning/models/eherkenning.py", line 170, in EherkenningConfiguration
        class Meta:
      File "/home/runner/work/django-digid-eherkenning/django-digid-eherkenning/digid_eherkenning/models/eherkenning.py", line 173, in Meta
        models.constraints.CheckConstraint(
    TypeError: CheckConstraint.__init__() got an unexpected keyword argument 'check'


The full traceback has been saved in:
/tmp/sphinx-err-35l5uhoy.log

To report this error to the developers, please open an issue at <https://github.com/sphinx-doc/sphinx/issues/>. Thanks!
Please also report this if it was a user error, so that a better error message can be provided next time.
make: *** [Makefile:20: html] Error 2
docs: exit 2 (0.80 seconds) /home/runner/work/django-digid-eherkenning/django-digid-eherkenning/docs> make SPHINXOPTS=-W html pid=3096
  docs: FAIL code 2 (20.31=setup[19.51]+cmd[0.80] seconds)
  evaluation failed :( (20.35 seconds)
```

The bug wasn't caught earlier because:
- The last CI run was August 4, 2025 (before Django 6.0 was released)
- Regular test environments pin to Django~=5.2, so they don't get Django 6.0
- Only the docs environment was affected due to lack of version ceiling

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -31,7 +31,7 @@ classifiers = [
 requires-python = ">=3.10"
 dependencies = [
     "cryptography>=40.0.0",
-    "django>=4.2.0",
+    "django>=4.2.0,<6.0.0",
     "django-sessionprofile",
     "django-simple-certmanager>=2.3.0",
     "django-solo",

